guides/plugins/plugins/storefront/add-dynamic-content-via-ajax-calls.md

```diff
@@ -0,0 +1,138 @@
+It combines and builds upon the the guides about [adding custom Javascript](./add-custom-javascript.md) and [adding a custom controller](./add-custom-controller.md), so you should probably read them first.
+
+## Setting up the Controller
+
+For this guide we will use a very simple controller that returns a timestamp wrapped in the JSON format.
+
+As mentioned before this guide builds up upon the [adding a custom controller](./add-custom-controller.md) guide.
+This means that this article will only cover the differences between returning a template and a `JSON` response and making it accessible to `XmlHttpRequests`.
+
+```php
+<?php declare(strict_types=1);
+
+namespace SwagBasicExample\Storefront\Controller;
+
+use Shopware\Core\Framework\Routing\Annotation\RouteScope;
+use Shopware\Storefront\Controller\StorefrontController;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\Routing\Annotation\Route;
+
+/**
+ * @RouteScope(scopes={"storefront"})
+ */
+class ExampleController extends StorefrontController
+{
+   /**
+    * @Route("/example", name="frontend.example.example", defaults={"XmlHttpRequest"=true}, methods={"GET"})
+    */
+    public function showExample(): JsonResponse
+    {
+        return new JsonResponse(['timestamp' => gmdate("Y-m-d\TH:i:s\Z")]);
+    }
+}
+```
+
+As you might have seen this controller isn't too different from the controller used in the before mentioned article.
+The route annotation has an added `defaults={"XmlHttpRequest"=true}` to allow XmlHttpRequest and it returns a `JsonResponse` instead of a normal `Response`.
+Using a `JsonResponse` instead of a normal `Response` causes the data structures passed to it to be automatically turned into a `JSON` string.
+
+The following `services.xml` and `routes.xml` are identical to before mentioned article, but here they are for reference anyways:
+
+```xml
+<?xml version="1.0" encoding="UTF-8" ?>
+<routes xmlns="http://symfony.com/schema/routing"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="http://symfony.com/schema/routing
+        https://symfony.com/schema/routing/routing-1.0.xsd">
+
+    <import resource="../../Storefront/Controller/**/*Controller.php" type="annotation" />
+</routes>
+```
+
+```xml
+<?xml version="1.0" ?>
+
+<container xmlns="http://symfony.com/schema/dic/services" 
+           xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+           xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+    <services>
+        <service id="SwagBasicExample\Storefront\Controller\ExampleController" public="true">
+            <call method="setContainer">
+                <argument type="service" id="service_container"/>
+            </call>
+        </service>
+    </services>
+</container>
+```
+
+## Preparing the Plugin
+
+Now we have to add a `plugin` to display the timestamp we get from our controller.
```

Comment:
This is not your fault, but i **HATE** the fact, that we called that javascript stuff "plugin".

Maybe write the following, to distinguish it better from the casual `plugin`, which is an extension to the whole shopware stack, to prevent some newcomers from being confused
```suggestion
Now we have to add a `Storefront (or maybe JavaScript?) plugin` to display the timestamp we get from our controller.
```

Reply:
```suggestion
Now we have to add a `Storefront Javascript plugin` to display the timestamp we get from our controller.
```
I'm calling it a Storefront Javascript plugin, now